src/preamble.js

```diff
@@ -295,8 +295,24 @@ function updateGlobalBufferAndViews(buf) {
   Module['HEAPF32'] = HEAPF32 = new Float32Array(buf);
   Module['HEAPF64'] = HEAPF64 = new Float64Array(buf);
 #if WASM_BIGINT
-  Module['HEAP64'] = HEAP64 = new BigInt64Array(buf);
-  Module['HEAPU64'] = HEAPU64 = new BigUint64Array(buf);
+    #if MIN_SAFARI_VERSION < 150000
+    // All browsers other than Safari added BigInt and BigInt64Array at the same
+    // time, but Safari introduced BigInt in v14.0 and introduced BigInt64Array
+    // in v15.0
+    if (typeof BigInt64Array === "undefined") {
+      #include "polyfill/bigint64array.js"
+      Module['HEAP64'] = HEAP64 = createBigInt64Array(HEAPU32);
+      Module['HEAPU64'] = HEAPU64 = createBigUint64Array(HEAPU32);
+    } else {
+      Module['HEAP64'] = HEAP64 = new BigInt64Array(buf);
+      Module['HEAPU64'] = HEAPU64 = new BigUint64Array(buf);
+    }
+    #else
+    Module['HEAP64'] = HEAP64 = new BigInt64Array(buf);
+    Module['HEAPU64'] = HEAPU64 = new BigUint64Array(buf);
+    #endif
+  }
```

Comment:
This closing brace looks like it needs to be removed.

Can you un-indent all the lines above by one level.. and keep the #if/#else/#endif in column 0?